src/ProxyKit/UpstreamHost.cs

```diff
@@ -51,19 +52,14 @@ private Uri GetUri()
 
         private int DefaultPort(string scheme)
         {
-            switch (scheme.ToLower())
+            return scheme.ToLowerInvariant() switch
```

Comment:
should be using String.Equals(...,..., StringComparison.OrdinalIgnoreCase) rather than triggering copying/allocation for no reward

Reply:
Even better, we can rely on this being lowercase by normalizing the scheme during construction. Uri().Scheme already normalizes for us.